File: src/cpp_pkg/src/string_utils.cpp
#include "string_utils.hpp"
#include <algorithm>

namespace string_utils {

std::string shout(const std::string &text) {
  std::string result = text;
  std::transform(result.begin(), result.end(), result.begin(), ::toupper);
  return result;
}

std::string Formatter::surround(const std::string &text,
                                const std::string &symbol) {
  return symbol + text + symbol;
}

} // namespace string_utils

File: src/py_pkg/string_utils.py
"""
math_utils.py - A dummy module for demonstration
"""

DEFAULT_GREETING = "Hello"


def shout(text):
    """
    Convert text to uppercase.

    Parameters
    ----------
    text : str
        The input string to convert.

    Returns
    -------
    str
        The uppercase version of the input string.
    """
    return text.upper()


class Formatter:
    """
    A class for formatting strings.
    """

    def surround(self, text, symbol="*"):
        """
        Surround text with a given symbol.

        Parameters
        ----------
        text : str
            The input string to surround.
        symbol : str, optional
            The symbol to use for surrounding (default is '*').

        Returns
        -------
        str
            The formatted string with symbols surrounding the text.
        """
        return f"{symbol}{text}{symbol}"
